crates/sui-types/src/error.rs

```diff
@@ -702,6 +702,18 @@ pub enum SuiError {
 
     #[error("Nitro attestation verify failed: {0}")]
     NitroAttestationFailedToVerify(String),
+
+    #[error("Failed to serialize {type_info:?}, error: {error:?}")]
+    GrpcMessageSerializeError { type_info: String, error: String },
+
+    #[error("Failed to deserialize {type_info:?}, error: {error:?}")]
+    GrpcMessageDeserializeError { type_info: String, error: String },
+
+    #[error("Transaction position is too far ahead: {round:?}, last committed round: {last_committed_round:?}")]
+    InvalidTransactionPosition {
```

Comment:
Since this name and message are primarily consumed by the caller, when this happens usually it means the validator is lagging, maybe call this `LaggingConsensusCommits`? And if a client indeed sends an invalid `ConsensusTxPosition` in a request, we don't care what error gets returned to them anyway.